test/prototype_common_utils.py

```diff
@@ -178,22 +178,58 @@ def make_one_hot_labels(
         yield make_one_hot_label(extra_dims_)
 
 
-def make_segmentation_mask(size=None, *, num_objects=None, extra_dims=(), dtype=torch.uint8):
+def make_detection_mask(size=None, *, num_objects=None, extra_dims=(), dtype=torch.uint8):
```

Comment:
The naming is now quite weird. We are calling a `make_detection_mask` function and get a `features.SegmentationMask` back. Now that we have somewhat settled on the terms segmentation and detection masks, can we maybe rename the class to `features.Mask` since it represents both types?

Reply:
Sounds good to me!

Reply:
On a second thought, I remember @YosuaMichael telling us that on Depth Perception they also have a sort of Valid Masks (see https://github.com/pytorch/vision/issues/6495).

@YosuaMichael could you clarify if they operate exactly like Segmentation/Detection masks and if we could just reuse this object?

Reply:
To give more context:

- segmentation mask is of shape `(*, H, W)` where each value corresponds to a class index
- detection mask is of shape `(*, N, H, W)` where `N` denotes the number of objects and the values are otherwise "boolean"

In both cases the `*` in the shape denotes arbitrary batch dimensions

Reply:
@datumbox I think the valid_mask on depth perception will behave like detection_mask with 2 object (it is boolean, but can be implemented with 0/1 uint as well). So overall I think we can reuse it.